lib/modules/dosomething/dosomething_signup/dosomething_signup.module

```diff
@@ -581,8 +581,33 @@ function dosomething_signup_user_signup($nid, $account = NULL, $source = NULL) {
       return;
     }
 
-    // Set default signup message if we're still in this function.
-    dosomething_signup_set_takeover();
+    $existing_user = dosomething_signup_not_first_signup($account->uid);
+
+    // Set signup message, depending on type of user with specific params.
+    if ($existing_user) {
+      // pass repeat-signup-existing in source param to
+      dosomething_signup_set_signup_message();
+    }
+    else {
+      // pass repeat-signup-new in source param to
+      dosomething_signup_set_takeover();
+    }
+  }
+}
+
+/**
+ * Returns true if user this is not a user's first signup.
+ * @param int $user uid
+ */
+function dosomething_signup_not_first_signup($uid) {
```

Comment:
This might be a little nitpicky - I feel like this should be flipped like `dosomething_signup_first_signup()` and return `true` if this is the users first sign up. 

I think it would make for more readable code to have to not check for the false positive when checking if this is the users first sign up. So you can do...

```
if (dosomething_signup_first_signup) {
dosomething interesting and different.
} else {
dosomething else. 
}
```